designs/kwok-provider.md

```diff
@@ -0,0 +1,51 @@
+# Karpenter KwoK Cloud Provider - v0.34+
+
+The kubernetes-sigs/karpenter repo uses a simulated kubernetes environment to test code changes. Users must use a cloud provider of their choice to test their changes, which can be costly and creates an unreasonable barrier to entry for contributors. Related issue: https://github.com/kubernetes-sigs/karpenter/issues/895.
+
+Creating a Cloud Provider neutral implementation of Karpenter within Karpenter-core requires creating a set of CP neutral examples of instance types and pricing that captures common tenets across CPs. 
+
+This is meant to be a living document to in-depth go through the decisions on how kubernetes-sigs/karpenter models its fake instance types. 
+
+## Proposal
+
+### Instance Types
+The full set of instance types for the KwoK Cloud Provider should be at least the cartesian product of the following dimensions, generating 60 instance types across 4 zones, totaling `12 * 3 * 2 = 64` in-memory Cloud Provider instance types.
+
+```md
+1. Instance Type Name: (family)-(size)
+    a. size: 1x, 2x, 4x, 8x, 16x, 32x, 48x, 64x, 96x, 128x, 192x, 256x
```

Comment:
```suggestion
    a. size: 1x, 2x, 4x, 9x, 32x, 100x, 128x, 256x
```
This deliberately doesn't match any particular cloud provider's size ratios, but does allow testing the logic side independent of what the real-world ratios might be (you can always find a node set to provide capacity for a workload).

If we accidentally coded in a power-of-two assumption, the 9x side ought to help pick that up.

Reply:
Good point. I'll make the change to add in 9x. Any reason you think it should replace the 32x and the same for 100x for 48x, 64x, and 96x?